server/service/apple_mdm.go

```diff
@@ -6815,7 +6855,10 @@ func (svc *Service) GetOTAProfile(ctx context.Context, enrollSecret string) ([]b
 		return nil, ctxerr.Wrap(ctx, err, "getting app config to get org name")
 	}
 
-	profBytes, err := apple_mdm.GenerateOTAEnrollmentProfileMobileconfig(cfg.OrgInfo.OrgName, cfg.MDMUrl(), enrollSecret)
+	// TODO(IB): Validate that the IdpUUID should be populated based on the criteria for showing the SSO in the first place
+	// Should be added with the work of #30660 or afterwars.
```

Comment:
This makes me think, if you have the enroll secret and the IDP account UUID, you can forge an enrollment profile that would get associated with this account, without going through the SSO authentication. Should we add some salt/state associated with the SAML response + IDP account so that the cookie does not contain _just_ the IDP Account UUID? We may need a security expert eye on this.

Reply:
You are right, one could definitely do that.

I can't speak to the issues of this in terms of security as it would require one to know both the enroll secret and a specific IDP UUID, which to me seems unlikely to know both, but again unsure who to ask and who can deem it a requirement for more strict approach.

Reply:
> Should we add some salt/state associated with the SAML response + IDP account so that the cookie does not contain just the IDP Account UUID? We may need a security expert eye on this.

@JordanMontgomery, I think we discussed something along these lines during estimation? IIRC the idea was to issue a one-time, expiring challenge (perhaps an extension of `datastore/challenges`) tied to the login session.